.github/workflows/wait-for-connection-on-error-test.yaml

```diff
@@ -0,0 +1,34 @@
+# A workflow to test waiting for SSH connections to the runner, when an error occurs.
+name: Connection Test - On Error
+on:
+  pull_request:
+    paths:
+    - ci_connection
+    - .github/workflows/wait-for-connection-on-error-test.yaml
+    branches:
+      - main
+      - wait_connect_on_error
+# Cancel any previous iterations if a new commit is pushed
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+jobs:
+  wait-on-error-test:
+    strategy:
+      fail-fast: false
+      matrix:
+        runner: ["linux-x86-n2-16"]
+        instances: ["1"]
+    runs-on: ${{ matrix.runner }}
+    timeout-minutes: 60
+    container:
+      image: us-central1-docker.pkg.dev/tensorflow-sigs/tensorflow/ml-build@sha256:cec01011e627c0fd101c521a8af7c09ce4557ab6dcc9f8678aeb67a3182ed821 # ratchet:tensorflow/ml-build:latest
+    steps:
+      - uses: actions/checkout@eef61447b9ff4aafe5dcd4e0bbf5d482be7e7871 # ratchet:actions/checkout@v4
+      - name: Echo
+        shell: bash  # required as the default `sh` doesn't have the ERR status
```

Comment:
It defaults to `sh`. I made it a default for the entire workflow, but it's still needed:
https://docs.github.com/en/actions/writing-workflows/choosing-where-your-workflow-runs/running-jobs-in-a-container#overview

Reply:
Bash being the chosen one is fine(and expected) the comment on why its being changed isn't really necessary as we typically change to bash for a multitude of reasons in various workflows.

Reply:
Got it. I would still rather keep the comment. I think that just like for me, it may be unexpected for most people that `sh` is used by default, and without the comment, it's a little easier to miss that this is required. Even in terms of general awareness of the shell used by default, I think the comment is a net benefit

Reply:
Fair enough